### PR TITLE
feat: Support for EC P-256 Key

### LINKED ISSUE #145: Issuer and Verifiers APIs - Support for NIST P256 version
Issuer and Verifiers need to support NIST P256

### PR DESCRIPTION
**Pull request by rolsonquadras**
closes #145 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: pkg/restapi/vc/operation/operations.go [hunk 1]
```diff
@@ -1490,13 +1499,57 @@ func (o *Operation) retrieveVC(profileName, docID, contextErrText string) ([]byt
...
+
+		encodedPrivateKey, err := x509.MarshalECPrivateKey(privateKey)
+		if err != nil {
+			return nil, "", err
+		}
+
+		// save the private key in DID as KMS only supports ed25519 keys
+		didPrivateKey = base58.Encode(encodedPrivateKey)
+
+		publicKey = &didclient.PublicKey{ID: pubKey1, Type: didclient.JWSVerificationKey2020,
```
Comment (fqutishat) [inline pkg/restapi/vc/operation/operations.go:1536]: why we just add two keys one for P256KeyType and other for Ed25519KeyType

Comment (fqutishat) [reply pkg/restapi/vc/operation/operations.go:1536]: pubKey1 for Ed25519KeyType.  pubKey2 for ECKeyType

File: test/bdd/features/vc_e2e_api.feature [hunk 2]
```diff
@@ -21,12 +21,15 @@ Feature: Using VC REST API
...
+      | myprofilewithdidsov  | university_degree.json       |                                                                  |                                                                                          | JWS             |       {"driverURL":"https://uniregistrar.io/1.0/register?driver-universalregistrar/driver-did-sov","options": {"network":"danube"}}                           | did:sov:danube  |    Ed25519Signature2018 |           |
+      | myprofile_ed25519    | university_degree.json       |                                                                  |                                                                                          | JWS             |                                                                                                                                                               | did:trustbloc   |   JsonWebSignature2020  |  Ed25519  |
```
Comment (fqutishat) [inline test/bdd/features/vc_e2e_api.feature:30]: can we use current profiles [myprofile_ud,myprofile_prc ]

**Comment by fqutishat** [comment]
@rolsonquadras did you run make bdd-interop-test

File: pkg/restapi/vc/operation/operations.go (hunk 1 above)
Comment (rolsonquadras) [reply pkg/restapi/vc/operation/operations.go:1536]: You mean always add 2 keys ?

**Comment by rolsonquadras** [comment]
> @rolsonquadras did you run make bdd-interop-test

@fqutishat Yes.

File: test/bdd/features/vc_e2e_api.feature (hunk 2 above)
Comment (rolsonquadras) [reply test/bdd/features/vc_e2e_api.feature:30]: sure.

File: pkg/restapi/vc/operation/operations.go (hunk 1 above)
Comment (fqutishat) [reply pkg/restapi/vc/operation/operations.go:1536]: yes

File: test/bdd/features/vc_e2e_api.feature (hunk 2 above)
Comment (fqutishat) [reply test/bdd/features/vc_e2e_api.feature:30]: myprofile_ud for Ed25519   and myprofile_prc  for P256

File: pkg/restapi/vc/operation/operations.go (hunk 1 above)
Comment (fqutishat) [reply pkg/restapi/vc/operation/operations.go:1536]: we can change it later. but the plan that we need did doc to include multiple keys

File: test/bdd/features/vc_e2e_api.feature (hunk 2 above)
Comment (troyronda) [reply test/bdd/features/vc_e2e_api.feature:30]: we also need a prc profile with ed25519. need to show both. @rolsonquadras @fqutishat

File: pkg/restapi/vc/operation/operations.go (hunk 1 above)
Comment (troyronda) [reply pkg/restapi/vc/operation/operations.go:1536]: yeh we should also show DID with multiple keys.

Comment (rolsonquadras) [reply pkg/restapi/vc/operation/operations.go:1536]: created an issue https://github.com/trustbloc/edge-service/issues/300

File: test/bdd/features/vc_e2e_api.feature (hunk 2 above)
Comment (rolsonquadras) [reply test/bdd/features/vc_e2e_api.feature]: [Added](https://github.com/trustbloc/edge-service/compare/2930edf978b9485a9e844730946e04a38e4817ce..4cd60a8250724ba9440a3a9bc740599c492d61e4#diff-d0c4c7ca6e3fe2063f6b701d16489b4bR30-R34) profiles for UD/PRC with ed25519/P256.

**Comment by rolsonquadras** [comment]
Patch code coverage is at 80%. Not able to increase as it's hard to generate error from crypto packages.

File: test/bdd/features/vc_e2e_api.feature [hunk 3]
```diff
@@ -21,12 +21,17 @@ Feature: Using VC REST API
...
+      | myprofilewithdidsov     | university_degree.json       |                                                                  |                                                                                          | JWS             |       {"driverURL":"https://uniregistrar.io/1.0/register?driver-universalregistrar/driver-did-sov","options": {"network":"danube"}}                           | did:sov:danube  |    Ed25519Signature2018 |           |
+      | myprofile_ud_ed25519    | university_degree.json       |                                                                  |                                                                                          | JWS             |                                                                                                                                                               | did:trustbloc   |   JsonWebSignature2020  |  Ed25519  |
```
Comment (fqutishat) [inline test/bdd/features/vc_e2e_api.feature:30]: this profile like myprofile_ud

File: test/bdd/features/vc_e2e_api.feature [hunk 4]
```diff
@@ -21,12 +21,17 @@ Feature: Using VC REST API
...
+      | myprofile_ud_p256_jws   | university_degree.json       |                                                                  |                                                                                          | JWS             |                                                                                                                                                               | did:trustbloc   |   JsonWebSignature2020  |  P256     |
```
Comment (fqutishat) [inline test/bdd/features/vc_e2e_api.feature:31]: wyu just add P256    to keyType  in myprofile_prc

Comment (fqutishat) [reply test/bdd/features/vc_e2e_api.feature:31]: it's already empty

File: test/bdd/features/vc_e2e_api.feature [hunk 5]
```diff
@@ -21,12 +21,17 @@ Feature: Using VC REST API
...
+      | myprofile_ud_p256_pv    | university_degree.json       |                                                                  |                                                                                          | ProofValue      |                                                                                                                                                               | did:trustbloc   |   JsonWebSignature2020  |  P256     |
+      | myprofile_prc_ed25519   | permanent_resident_card.json |                                                                  |                                                                                          | JWS             |                                                                                                                                                               | did:trustbloc   |   JsonWebSignature2020  |  Ed25519  |
```
Comment (fqutishat) [inline test/bdd/features/vc_e2e_api.feature:33]: this profile like myprofile_prc

File: pkg/doc/vc/crypto/crypto.go [hunk 6]
```diff
@@ -78,12 +102,24 @@ func (s *kmsSigner) Sign(data []byte) ([]byte, error) {
...
-func newPrivateKeySigner(privateKey []byte) *privateKeySigner {
-	return &privateKeySigner{privateKey: privateKey}
+func newPrivateKeySigner(keyType string, privateKey []byte) *privateKeySigner {
+	return &privateKeySigner{keyType: keyType, privateKey: privateKey}
 }
 
 func (s *privateKeySigner) Sign(data []byte) ([]byte, error) {
-	return ed25519.Sign(s.privateKey, data), nil
+	switch s.keyType {
+	case "", Ed25519KeyType:
```
Comment (fqutishat) [inline pkg/doc/vc/crypto/crypto.go:111]: why we support keyType  is empty

Comment (sudeshrshetty) [reply pkg/doc/vc/crypto/crypto.go:111]: may be default? @rolsonquadras ^^

Comment (rolsonquadras) [reply pkg/doc/vc/crypto/crypto.go:111]: That't correct @sudeshrshetty.

File: test/bdd/features/vc_e2e_api.feature (hunk 3 above)
Comment (rolsonquadras) [reply test/bdd/features/vc_e2e_api.feature:30]: That't with uniRegistrar value. It's empty in this profile.

File: test/bdd/features/vc_e2e_api.feature (hunk 4 above)
Comment (rolsonquadras) [reply test/bdd/features/vc_e2e_api.feature:31]: That't with uniRegistrar value. It's empty in this profile.

File: test/bdd/features/vc_e2e_api.feature (hunk 5 above)
Comment (rolsonquadras) [reply test/bdd/features/vc_e2e_api.feature:33]: That't with uniRegistrar value. It's empty in this profile.

**Pull request merged**